design/bgp-bfd.md

```diff
@@ -0,0 +1,164 @@
+# BFD enablement in BGP
+
+## Summary
+
+The purpose of this enhancement is to use enable [BFD](https://en.wikipedia.org/wiki/Bidirectional_Forwarding_Detection) to be used in conjunction with BGP.
+
+BFD will be available only in the upcoming [FRR based implementation](0001-frr.md).
+
+## Motivation
+
+BFD provides a quicker path failure detection than BGP, and using them together will allow users to provide a better
+service.
+
+BFD for BGP is supported by FRR [out of the box](http://docs.frrouting.org/en/latest/bfd.html#bgp-bfd-configuration).
+
+### Goals
+
+* Provide a way to establish BFD on BGP session.
+* Provide a way to configure the parameters of the BFD session associated with a BGP session.
+
+### Non-Goals
+
+* Implement BFD on the legacy BGP implementation.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a cluster administrator, I want to declare a BGP session to be backed up by BFD.
+
+#### Story 2
+
+As a cluster administrator, I want to be able to set the BFD parameters related to a BGP session.
+
+## Design Details
```

Comment:
Can you add a comment somewhere about how MetalLB will behave if this gets configured with the legacy BGP stack in use?

Reply:
Will do